test/miscellaneous/async-await.js

```diff
@@ -700,22 +768,45 @@ suite('async', () => {
       },
     });
 
+  testParse('({ async *a(){} })', expr,
+    {
+      type: 'ObjectExpression',
+      properties: [
+        {
+          type: 'Method',
+          name: {
+            type: 'StaticPropertyName',
+            value: 'a',
+          },
+          isAsync: true,
+          isGenerator: true,
+          params: {
+            type: 'FormalParameters',
+            rest: null,
+            items: [],
+          },
+          body: {
+            'directives': [],
+            'statements': [],
+            'type': 'FunctionBody',
+          },
+        },
+      ],
+    });
+
   suite('failures', () => {
     testParseFailure('async (a, ...b, ...c) => {}', ErrorMessages.UNEXPECTED_TOKEN(','));
     testParseFailure('async\n(a, b) => {}', 'Unexpected token "=>"');
     testParseFailure('new async() => {}', 'Unexpected token "=>"');
     testParseFailure('({ async\nf(){} })', 'Unexpected identifier');
-    testParseFailure('async ((a)) => {}', 'Unexpected token "=>"');
+    testParseFailure('async ((a)) => {}', ErrorMessages.UNEXPECTED_TOKEN('=>'));
     testParseFailure('({ async get a(){} })', 'Unexpected identifier');
     testParseFailure('async a => {} ()', 'Unexpected token "("');
     testParseFailure('a + async b => {}', 'Unexpected token "=>"');
     testParseFailure('a + async () => {}', 'Unexpected token "=>"');
     testParseFailure('with({}) async function f(){};', 'Unexpected token "function"');
     testParseFailure('function* a(){ async yield => {}; }', '"yield" may not be used as an identifier in this context');
     testParseFailure('function* a(){ async (yield) => {}; }', 'Unexpected token "=>"');
-    testParseFailure('async function* a(){}', 'Unexpected token "*"');
-    testParseFailure('(async function* (){})', 'Unexpected token "*"');
```

Comment:
This test and the one above it should be asserted to parse, rather than removed outright.